sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/client/DefaultHttpClient.java

```diff
@@ -192,39 +196,214 @@ private void sendBody(HttpRequest httpRequest, Object progressReporter, HttpURLC
+    private static AccessibleByteArrayOutputStream getAccessibleByteArrayOutputStream(HttpURLConnection connection) throws IOException {
+        AccessibleByteArrayOutputStream outputStream = new AccessibleByteArrayOutputStream();
 
-            for (Map.Entry<String, List<String>> entry : connection.getHeaderFields().entrySet()) {
-                if (entry.getKey() != null) {
-                    responseHeaders.add(HeaderName.fromString(entry.getKey()), entry.getValue());
+        try (InputStream errorStream = connection.getErrorStream();
+             InputStream inputStream = (errorStream == null) ? connection.getInputStream() : errorStream) {
+            byte[] buffer = new byte[8192];
+            int length;
+            while ((length = inputStream.read(buffer)) != -1) {
+                outputStream.write(buffer, 0, length);
+            }
+        }
+        return outputStream;
+    }
+
+    /**
+     * Processes the sse buffer and dispatches the event
+     *
+     * @param reader   The BufferedReader object
+     * @param listener The listener object attached with the httpRequest
+     */
+    private RetrySSEResult processBuffer(BufferedReader reader, ServerSentEventListener listener) {
+        StringBuilder collectedData = new StringBuilder();
+        ServerSentEvent event = null;
+        try {
+            int dataRead = reader.read();
+            while (dataRead != -1) {
+                collectedData.append((char) dataRead);
+                dataRead = reader.read();
+                int index = isEndOfBlock(collectedData);
+                if (index >= 0) {
+                    String[] lines = collectedData.substring(0, index).split("\n"); // split the block into lines
+                    collectedData.delete(0, index + 2); // clear first block including "\n\n"
+                    event = processLines(lines);
+                    if (event != null
+                        && (!Objects.equals(event.getEvent(), DEFAULT_EVENT) || event.getData() != null)) {
+                        // dispatch the event if there is data or event
+                        listener.onEvent(event);
+                    }
                 }
             }
+            listener.onClose();
+        } catch (IOException e) {
+             return new RetrySSEResult(e, event.getId(), event.getRetryAfter());
+        }
+        return null;
+    }
+
+    private int isEndOfBlock(StringBuilder sb) {
+        // blocks of data are separated by double newlines
+        return sb.indexOf("\n\n");
+        // confirm the ways to check for end of block
+    }
+
+    private ServerSentEvent processLines(String[] lines) {
+        StringBuilder eventData = new StringBuilder();
+        ServerSentEvent event = new ServerSentEvent();
+
+        //start parsing line by line
+        for (String line : lines) {
+            int idx = line.indexOf(':');
+            if (idx < 0) {
+                continue; // ignore invalid data
```

Comment:
We should throw here instead of ignoring invalid data.

Reply:
It is not invalid data at this point. An event stream may or may not contain a comment line. `:` is indicating a comment line.